File: templates/templates/ruby/tests/custom-http/auth_test.rb
# frozen_string_literal: true
# typed: true

require_relative "../lib/customhttp"
require_relative "common_helper_test"

require "minitest/autorun"
require "minitest/focus"

module CustomHttp
  class TestAuth < Minitest::Test
    def setup
      @sdk = OpenApiSDK::SDK.new(server_url: CommonHelpers.api_test_service_url)
    end

    def test_custom_http_scheme_only
      record_test("auth-custom-security-scheme-only")

      test_scopes = ["read:products", "write:products"]

      @sdk = OpenApiSDK::SDK.new(
        server_url: CommonHelpers.api_test_service_url,
        custom_http: OpenApiSDK::Models::Components::SchemeCustomHTTPSecurity.new(
          user_id: 54_321,
          role: OpenApiSDK::Models::Components::Role::MANAGER,
          passphrase: "secure-passphrase-123",
          access_code: 104,
          scopes: test_scopes
        )
      )

      res = @sdk.auth.custom_http_only

      refute_nil(res)
      refute_nil(res.object)
      assert_equal("access_granted", res.object.grant)
      assert_equal(test_scopes, res.object.scopes)
    end
  end
end

File: templates/templates/php/Tests/custom-http/AuthTest.php
declare(strict_types=1);

namespace OpenAPI\OpenAPI\Tests;

use OpenAPI\OpenAPI\Tests\CommonHelpers;
use OpenAPI\OpenAPI\Models\Components\SchemeCustomHTTPSecurity;
use OpenAPI\OpenAPI\Models\Components\Role;
use PHPUnit\Framework\TestCase;

final class AuthTest extends TestCase
{
    public function testCustomHttpSchemeOnly(): void
    {
        CommonHelpers::recordTest('auth-custom-security-scheme-only');

        $testScopes = ['read:products', 'write:products'];

        $sdk = \OpenAPI\OpenAPI\SDK::builder()
            ->setServerUrl(CommonHelpers::getApiTestServiceUrl())
            ->setSecurity(new SchemeCustomHTTPSecurity(
                userID: 54321,
                role: Role::Manager,
                passphrase: 'secure-passphrase-123',
                accessCode: 104,
                scopes: $testScopes
            ))
            ->build();

        $response = $sdk->auth->customHttpOnly();
        $this->assertNotNull($response);
        $this->assertNotNull($response->object);
        $this->assertEquals('access_granted', $response->object->grant);
        $this->assertEquals($testScopes, $response->object->scopes);
    }
}
